### PR TITLE
Exploratory changes for producing global vector tiles

### PR DESCRIPTION
**Pull request by lossyrob**

### CODE REVIEW

File: src/main/scala/vectorpipe/Clip.scala [hunk 1]
```diff
@@ -74,33 +74,39 @@ object Clip {
     centre.distanceToSegment(p1, p2) <= radius
 
   /** Naively clips Features to fit the given Extent. */
-  def byExtent[G <: Geometry, D](extent: Extent, f: Feature[G, D]): Option[Feature[Geometry, D]] = {
+  def byExtent[G <: Geometry, D](extent: Extent, f: Feature[G, D], contains: (Extent, G) => Boolean): Option[Feature[Geometry, D]] = {
```
Comment (lossyrob) [inline src/main/scala/vectorpipe/Clip.scala]: Hacked in a `contains` to work around having the prepared geometry contains check only knowable from the call site.

File: src/main/scala/vectorpipe/Collate.scala [hunk 2]
```diff
@@ -120,24 +120,24 @@ object Collate {
...
 
       /* It was either a Way that became a Line/Poly, or a Relation that became a MultiPoly */
-      case ElementData(_, _, Some(Left(extent))) => {
+      case ElementData(_, _, None) => {
 
         /* Combine data for Nodes which were shared by multiple Ways */
         val kids = uniqueNodes(d.children.flatMap(t => work(d.root.meta.id, t)))
 
-        ((GeomExtra(extent), d.root) +: kids)
+        ((NoExtra, d.root) +: kids)
```
Comment (lossyrob) [inline src/main/scala/vectorpipe/Collate.scala]: Got rid of holding lat lngs and extents in metadata; this is bloated and doesn't need to happen - geometry reconstruction needs other mechanisms to solve other use cases (like changeset geometry reconstruction).

File: src/main/scala/vectorpipe/VectorPipe.scala [hunk 3]
```diff
@@ -129,43 +132,251 @@ object VectorPipe {
...
     }
+
+    multiPolygon.foreach(rasterExtent, options)(fn)
+    tiles.keys.asScala.map { case (col, row) => SpatialKey(col, row) }
   }
 
+
   /* Takes advantage of the fact that most Geoms are small, and only occur in one Tile */
-  private def byIntersect[G <: Geometry, D](
+  def byIntersect[G <: Geometry, D](
```
Comment (lossyrob) [inline src/main/scala/vectorpipe/VectorPipe.scala]: A lot of optimization here, also it is still slow for crazy geoms

File: src/main/scala/vectorpipe/osm/package.scala [hunk 4]
```diff
@@ -48,7 +48,13 @@ package object osm {
...
     * to properly configure Hadoop to read your S3 credentials.
     */
-  def fromORC(path: String)(implicit ss: SparkSession): Either[String, (RDD[Node], RDD[Way], RDD[Relation])] = {
+  def fromORC(path: String)(implicit ss: SparkSession): Either[String, (RDD[(Long, Node)], RDD[(Long, Way)], RDD[Relation])] =
+    Try(ss.read.orc(path)) match {
+      case Failure(e) => Left(e.toString)
+      case Success(data) => fromDataFrame(data)
+    }
+
+  def fromDataFrame(df: DataFrame)(implicit ss: SparkSession): Either[String, (RDD[(Long, Node)], RDD[(Long, Way)], RDD[Relation])] = {
```
Comment (lossyrob) [inline src/main/scala/vectorpipe/osm/package.scala]: Let us pass DataFrames in!

File: src/main/scala/vectorpipe/VectorPipe.scala [hunk 5]
```diff
@@ -129,43 +132,251 @@ object VectorPipe {
...
+    val extent = multiLine.envelope
+    val bounds: GridBounds = mt(extent)
+    val options = Rasterizer.Options(includePartial=true, sampleType=PixelIsArea)
 
-      /* Clip each geometry in some way */
-      val clipped: List[Feature[G, D]] = iter.foldLeft(List.empty[Feature[G, D]]) { (acc, g) =>
-        clip(kExt, g) match {
-          case Some(h) => h :: acc
-          case None => logError(kExt, g); acc /* Sneaky IO to log the error */
-        }
```
Comment (fosskers) [inline src/main/scala/vectorpipe/VectorPipe.scala]: I guess this was too naive?

File: src/main/scala/vectorpipe/osm/package.scala (hunk 4 above)
Comment (fosskers) [reply src/main/scala/vectorpipe/osm/package.scala]: Score.

**Comment by lossyrob** [comment]
Closing out.

**Pull request closed without merging**